src/world.rs

```diff
@@ -156,6 +213,7 @@ impl World {
 
             self.move_(ent.id, meta.ty, new_ty, meta.idx);
 
+            // TODO: Why use an Option if is always return Some<Bundle>? What happens if a component is not present?
```

Comment:
There is an early return a few lines above
```rust
B::remove(&mut types)?;
```
which returns `None` if one the components in the given bundle is not present.